diff --git a/index.js b/index.js
index a27f81ce041e7cf9b7dd9d95533813527f918fb6..daa176639b8449966dc295b298bbe08366b88869 100644
--- a/index.js
+++ b/index.js
@@ -104,7 +104,7 @@ function expand(str, isTop) {
     ? expand(m.post, false)
     : [''];
 
-  if (/\$$/.test(m.pre)) {    
+  if (m.pre.endsWith('\u0024' /*'$'*/)) {
     for (var k = 0; k < post.length; k++) {
       var expansion = pre+ '{' + m.body + '}' + post[k];
       expansions.push(expansion);
