diff --git a/index.js b/index.js
index 1234567890123456789012345678901234567890..1234567890123456789012345678901234567890 100644
--- a/index.js
+++ b/index.js
@@ -103,9 +103,9 @@
   var post = m.post.length
     ? expand(m.post, false)
     : [''];
 
-  if (/\$$/.test(m.pre)) {    
+  if (m.pre.endsWith('\u0024' /*'$'*/)) {
     for (var k = 0; k < post.length; k++) {
       var expansion = pre+ '{' + m.body + '}' + post[k];
       expansions.push(expansion);
     }
