diff --git a/dist/commonjs/index.js b/dist/commonjs/index.js
index 64a0f1f833222e446dcc0fbd59a04aca2328d532..924777527e9ccdeabb7bf6fbab69f0f862d44813 100644
--- a/dist/commonjs/index.js
+++ b/dist/commonjs/index.js
@@ -650,7 +650,8 @@ class Minimatch {
         }
         this.debug('matchOne', this, { file, pattern });
         this.debug('matchOne', file.length, pattern.length);
-        for (var fi = 0, pi = 0, fl = file.length, pl = pattern.length; fi < fl && pi < pl; fi++, pi++) {
+        var fi = 0, pi = 0, fl = file.length, pl = pattern.length;
+        for (; fi < fl && pi < pl; fi++, pi++) {
             this.debug('matchOne loop');
             var p = pattern[pi];
             var f = file[fi];
diff --git a/dist/esm/index.js b/dist/esm/index.js
index 84b577b0472cb6cb2fe4d5f2a146867fc5bdd8c5..85c0d52b4de726aaef68490b3165d5b517b4a675 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -638,7 +638,8 @@ export class Minimatch {
         }
         this.debug('matchOne', this, { file, pattern });
         this.debug('matchOne', file.length, pattern.length);
-        for (var fi = 0, pi = 0, fl = file.length, pl = pattern.length; fi < fl && pi < pl; fi++, pi++) {
+        var fi = 0, pi = 0, fl = file.length, pl = pattern.length;
+        for (; fi < fl && pi < pl; fi++, pi++) {
             this.debug('matchOne loop');
             var p = pattern[pi];
             var f = file[fi];
