diff --git a/dist/commonjs/index.js b/dist/commonjs/index.js
index 1234567890123456789012345678901234567890..1234567890123456789012345678901234567890 100644
--- a/dist/commonjs/index.js
+++ b/dist/commonjs/index.js
@@ -649,9 +649,10 @@
             file = this.levelTwoFileOptimize(file);
         }
         this.debug('matchOne', this, { file, pattern });
         this.debug('matchOne', file.length, pattern.length);
-        for (var fi = 0, pi = 0, fl = file.length, pl = pattern.length; fi < fl && pi < pl; fi++, pi++) {
+        var fi = 0, pi = 0, fl = file.length, pl = pattern.length;
+        for (; fi < fl && pi < pl; fi++, pi++) {
             this.debug('matchOne loop');
             var p = pattern[pi];
             var f = file[fi];
             this.debug(pattern, p, f);
@@ -1013,5 +1014,5 @@
 exports.minimatch.AST = ast_js_1.AST;
 exports.minimatch.Minimatch = Minimatch;
 exports.minimatch.escape = escape_js_1.escape;
 exports.minimatch.unescape = unescape_js_1.unescape;
-//# sourceMappingURL=index.js.map
\ No newline at end of file
+//# sourceMappingURL=index.js.map
diff --git a/dist/esm/index.js b/dist/esm/index.js
index 1234567890123456789012345678901234567890..1234567890123456789012345678901234567890 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -637,9 +637,10 @@
             file = this.levelTwoFileOptimize(file);
         }
         this.debug('matchOne', this, { file, pattern });
         this.debug('matchOne', file.length, pattern.length);
-        for (var fi = 0, pi = 0, fl = file.length, pl = pattern.length; fi < fl && pi < pl; fi++, pi++) {
+        var fi = 0, pi = 0, fl = file.length, pl = pattern.length;
+        for (; fi < fl && pi < pl; fi++, pi++) {
             this.debug('matchOne loop');
             var p = pattern[pi];
             var f = file[fi];
             this.debug(pattern, p, f);
@@ -997,5 +998,5 @@
 minimatch.AST = AST;
 minimatch.Minimatch = Minimatch;
 minimatch.escape = escape;
 minimatch.unescape = unescape;
-//# sourceMappingURL=index.js.map
\ No newline at end of file
+//# sourceMappingURL=index.js.map
